conn.go

```diff
@@ -15,9 +17,16 @@ const (
 )
 
 // Conn is a wrapper for the actual TLS connections made to Apple
-type Conn struct {
-	NetConn net.Conn
-	Conf    *tls.Config
+type Conn interface {
```

Comment:
This interface seems to include a number of concepts compared to interfaces I've seen in the standard library. Maybe it can be split up in some way?